go/vt/mysqlctl/mysqlctlproto/backup.go

```diff
@@ -17,15 +17,46 @@ limitations under the License.
 package mysqlctlproto
 
 import (
+	"strings"
+	"time"
+
+	"vitess.io/vitess/go/protoutil"
+	"vitess.io/vitess/go/vt/log"
+	"vitess.io/vitess/go/vt/mysqlctl"
 	"vitess.io/vitess/go/vt/mysqlctl/backupstorage"
+	"vitess.io/vitess/go/vt/topo/topoproto"
 
 	mysqlctlpb "vitess.io/vitess/go/vt/proto/mysqlctl"
 )
 
 // BackupHandleToProto returns a BackupInfo proto from a BackupHandle.
 func BackupHandleToProto(bh backupstorage.BackupHandle) *mysqlctlpb.BackupInfo {
-	return &mysqlctlpb.BackupInfo{
+	bi := &mysqlctlpb.BackupInfo{
 		Name:      bh.Name(),
 		Directory: bh.Directory(),
 	}
+
+	if parts := strings.Split(bi.Name, "."); len(parts) == 3 {
```

Comment:
Non-blocking (scope-creepy) suggestion: colocating this backup name parse logic with the format call that happens in [mysqlctl/backup.go](https://github.com/vitessio/vitess/blob/main/go/vt/mysqlctl/backup.go#L99) could be useful for searchability (and any future refactors).